[201911][Mellanox] Add hw-mgmt patch to support SDK OFFLINE event handling during ISSU

keboliu opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx" or "resolves #xxxx"

Please provide the following information:
-->

**- Why I did it**

During ISSU, "mlxsw_minimal" driver still trying to access firmware, in some cases FW could return some wrong critical threshold value which will cause switch shutdown. 

**- How I did it**

In order to prevent "mlxsw_minimal" driver from accessing ASIC during ISSU, SDK will raise "OFFLINE" 'udev' event
at the early beginning of such flow. When this event is received, hw-managemnet will remove "mlxsw_minimal" driver.
There is no need to implement the opposite "ONLINE" event since this flow is ended up with "kexec".

**- How to verify it**

repeatedly perform warm reboot, make sure there is no switch shutdown occurred.

**- Which release branch to backport (provide reason below if selected)**

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

andriymoroz-mlnx commented on device/mellanox/x86_64-mlnx_msn4700-r0/default_sku:
```diff
@@ -0,0 +1 @@
+ACS-MSN3700 t1
```
37->47

keboliu replied:
fixed.

andriymoroz-mlnx commented on device/mellanox/x86_64-mlnx_msn4700-r0/ACS-MSN4700/qos.json.j2:
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/qos.json.j2
```
27->47

andriymoroz-mlnx replied:
or it the same?

keboliu replied:
they are the same.

andriymoroz-mlnx commented on device/mellanox/x86_64-mlnx_msn4700-r0/sensors.conf:
```diff
@@ -0,0 +1,116 @@
+################################################################################
+# Copyright (c) 2019 Mellanox Technologies
+#
+# Platform specific sensors config for SN3700
```
another 3700
also sensors list can be different. We probably don't know yet...

keboliu committed:
Add hw-mgmt patch to support SDK OFFLINE event for handling flow within service firmware upgrade

In order to prevent "mlxsw_minimal" driver accessing ASIC during in
service firmware upgrade flow, SDK will raise "OFFLINE" 'udev' event
at early beginning of such flow. When this event is received,
hw-managemnet will remove "mlxsw_minimal" driver.
There is no need to implement opposite "ONLINE" event, since this flow
is ended up with "kexec".

Signed-off-by: Kebo Liu <kebol@nvidia.com>
